core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -47,6 +47,19 @@
  * Metadata for a table.
  */
 public class TableMetadata implements Serializable {
+
+  /**
+   * Iceberg always use the latest stable format version when creating a new table.
+   * This reserved property keyword allows users to override the Iceberg format version of the table metadata.
+   * <p>
+   * If this table property exists when creating a table, the table will use the specified format version.
+   * If a table updates this property, it will try to upgrade to the specified format version.
+   * This helps developers to try out new features in an unreleased version or migrate existing tables to a new version.
+   * <p>
+   * Note: If you set the Iceberg format to an unreleased version, software stability is not guaranteed.
```

Comment:
This reserved property should not allow setting the format version to anything other than a stable and adopted version.